### PR TITLE
Add `kail` to the Prow testing image.

### PR DESCRIPTION
**Pull request by coryrc**
The `go get -u` command for it is broken, so using the recommended download command from the repo.

First half of solving https://github.com/knative/serving/issues/5459

Have not pushed image yet.

### CODE REVIEW

File: images/prow-tests/install-kail.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
```
Comment (chaodaiG) [inline images/prow-tests/install-kail.sh]:
```suggestion
#!/usr/bin/env bash
```
see this PR for reference: https://github.com/knative/test-infra/pull/1350

**Comment by chaodaiG** [comment]
/approve
/cc @adrcunha 
for a second pair of eyes

File: images/prow-tests/install-kail.sh (hunk 1 above)
Comment (coryrc) [reply images/prow-tests/install-kail.sh]: Oh boy, more brittleness

File: images/prow-tests/Dockerfile [hunk 2]
```diff
@@ -61,6 +61,9 @@ ADD . /go/src/knative.dev/test-infra
 RUN make -C /go/src/knative.dev/test-infra/tools/githubhelper
 RUN cp /go/src/knative.dev/test-infra/tools/githubhelper/githubhelper .
 
+# We would prefer to install with go get -u github.com/boz/kail, but it is broken
+RUN /go/src/knative.dev/test-infra/images/prow-tests/install-kail.sh
```
Comment (adrcunha) [inline images/prow-tests/Dockerfile]: I think you don't need an extra script for that. Please prove me wrong. Or not.

**Comment by adrcunha** [comment]
"go get -u" is not broken, it's a matter of mismatched dependencies.

```
$ go get -u github.com/boz/kail
/go/src/k8s.io/client-go/transport/round_trippers.go:70:11: cannot convert klog.V(9) (type klog.Verbose) to type bool
/go/src/k8s.io/client-go/transport/round_trippers.go:72:11: cannot convert klog.V(8) (type klog.Verbose) to type bool
/go/src/k8s.io/client-go/transport/round_trippers.go:74:11: cannot convert klog.V(7) (type klog.Verbose) to type bool
/go/src/k8s.io/client-go/transport/round_trippers.go:76:11: cannot convert klog.V(6) (type klog.Verbose) to type bool
```

https://github.com/kubernetes/client-go/issues/654

File: images/prow-tests/Dockerfile (hunk 2 above)
Comment (adrcunha) [reply images/prow-tests/Dockerfile]: Fix the dependencies, use "go get", see my comment in the main thread.

Comment (coryrc) [reply images/prow-tests/Dockerfile]: I already did in our chat

**Comment by adrcunha** [comment]
Obsoleted by #1378.

/close

**Pull request closed without merging**